zerver/lib/test_fixtures.py

```diff
@@ -123,18 +121,12 @@ def is_current(self) -> bool:
             'tools/setup/postgres-init-dev-db',
             'zerver/migrations/0258_enable_online_push_notifications_default.py',
         ]
-        check_settings = [
-            'REALM_INTERNAL_BOTS',
-        ]
```

Comment:
I don't want to remove the `check_settings` infrastructure; we're definitely going to end up needing to re-add it in the future.  (And I think there are probably a few settings that impact `populat_db` and probably we'll add to this list the next time we change them).

I like the direction of the refactor, though; maybe we can just add a new argument to `is_digest_obsolete` that's a dictionary (extracted from `settings`) of custom values to hash a stable marshalling (sort-then-json-dump?) together with the rest?

Reply:
Based on your input, I kept `REALM_INTERNAL_BOTS` and added two other bot-related settings which are now more likely to trigger a rebuild.